### PR TITLE
[Fix] Allow behavioural only processes

### LINKED ISSUE #9502: ✨ Take into account processes with ONLY behavioural skills on "Assessment plan"
## ✨ Feature

Right now when a brand new process ONLY includes behavioural skills, the assessment plan results in Step 1: Application show an error message saying skills are missing from this assessment, since this step is pre-populated and can only be used to assess technical skills.

This makes this error impossible to remove.

Technically we should allow processes with only behavioural skills, since the application is still used to assess other requirements (e.g. Education requirement).

## 🕵️ Details

We need to make sure we allow behavioural only processes.

We can add the "Education requirement" as a permanent item in the assessment card where we show the skills that are gonna be assessed by this assessment.

This way, we prevent this step from giving us an error, since it will always be part of the assessment plan to at least assess the education requirement.

## 🎨 Design File

[Figma file](https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Admin---Processes%2FPools%2FRecruitments---Record-of-Decision?type=design&node-id=8543%3A22364&mode=design&t=EUQea4foVB5VMzjo-1) by @Jerryescandon 

### Screenshot of error

Image

### Screenshot of new design

Image

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Assessment plan displays correctly (application with Education requirement, etc..) even if there are only behavioural skills on the job advertisement.

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #9502 

## 👋 Introduction

Updates process assessment plan validation to allow submitting plans with only behavioural skills.

## 🕵️ Details

I hope this solution was what we had in mind 😅 

## 🧪 Testing

1. Build `npm run dev`
2. Login as admin
3. Create a pool and complete the advertisement but do not add any technical skills
4. Go to the assessment plan
5. Confirm that there is no error on the first step (Application screening)
6. Confirm application screening shows education
7. Confirm you can complete the plan by only assessing behavioural skills
8. Confirm you can now publish the pool

## 📸 Screenshot

Screenshot 2024-03-05 121907
Screenshot 2024-03-05 121914

### CODE REVIEW

File: apps/web/src/validators/pool/assessmentPlan.ts [hunk 1]
```diff
@@ -26,9 +26,13 @@ export function getAssessmentPlanStatus(pool: Pool): PoolCompleteness {
...
 
-  // disregard screening question step for step validation
+  // disregard screening questions and application screening step for step validation
   const assessmentStepsWithoutScreeningQuestion = pool.assessmentSteps.filter(
-    (step) => step?.type !== AssessmentStepType.ScreeningQuestionsAtApplication,
+    (step) =>
+      ![
+        AssessmentStepType.ApplicationScreening,
+        AssessmentStepType.ScreeningQuestionsAtApplication,
+      ].includes(step?.type ?? AssessmentStepType.ApplicationScreening),
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/validators/pool/assessmentPlan.ts:35]: I think it is easy enough to do if this is the only place it needs to be touched, assuming you are correct. @tristan-orourke should I only be checking `ApplicationScreening` here?

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentStepCard.tsx [hunk 2]
```diff
@@ -99,16 +101,25 @@ const AssessmentStepCard = ({
...
           data-h2-padding-left="base(0)"
           data-h2-margin-top="base(x.5)"
         >
+          {isApplicationScreening && (
+            <li data-h2-padding-left="base(0)" data-h2-display="base(inline)">
+              {intl.formatMessage({
+                defaultMessage: "Education",
+                id: "Oi5hI5",
+                description: "Label for the education requirement screening",
+              })}
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentStepCard.tsx]: Oof, I totally misread that. God catch! Updated in [update term on assessment card](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9645/commits/33afddda39425af67c439fd8970b30d3bdd2af0d)

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentStepCard.tsx [hunk 3]
```diff
@@ -99,16 +101,25 @@ const AssessmentStepCard = ({
...
+            </li>
+          )}
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentStepCard.tsx:115]: Oops, confused my `&&` and `||` 😛  Fixed in [fix bullet separator for application screening step](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9645/commits/0dafbbe0a161960568b8eed037c07d95fea396f8)

File: apps/web/src/validators/pool/assessmentPlan.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply apps/web/src/validators/pool/assessmentPlan.ts:35]: @esizer correct. ScreeningQuestion steps now need skills. ApplicationScreening step does not.

[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/validators/pool/assessmentPlan.ts]: Thanks! Updated in [update validator to check screening questions assesses skills](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9645/commits/3ed7225b8bd55ef4b1a89d57c092dc8e0369ce5e)

**Review by vd1992 — APPROVED** [review]
Failing Jest test aside, good to go! Just think you need to update the string duplicates there, judging by which test failed. 

Nice!

**Pull request merged**